coda/coda_replication/tests/test_views.py

```diff
@@ -0,0 +1,496 @@
+        response = views.queue(request)
+        assert response.status_code == 200
+
+        xml_obj = objectify.fromstring(response.content)
+        assert len(xml_obj.entry) == 20
+
+    @pytest.mark.xfail(reason="0 entries should be returned.")
+    def test_get_without_identifier_correctly_filters_by_status(self, rf):
+        factories.QueueEntryFactory.create_batch(30, status=1)
+        request = rf.get('/', {'status': 2}, HTTP_HOST='example.com')
+        response = views.queue(request)
+        assert response.status_code == 200
+
+        xml_obj = objectify.fromstring(response.content)
+        for entry in xml_obj.entry:
+            queue = entry.content[QUEUE_ENTRY]
+            assert queue.status == 2
+
+    @pytest.mark.xfail(reason='This enables a path of execution where the '
+                              'queueEntries variable is accessed before it is set.')
+    def test_get_without_identifier_with_falsy_sort_parameter(self, rf):
+        factories.QueueEntryFactory.create_batch(30)
+        request = rf.get('/', {'sort': 0}, HTTP_HOST='example.com')
+        response = views.queue(request)
+        assert response.status_code == 200
+
+    @pytest.mark.xfail(reason='Empty Page exception thrown from the makeObjectFeed function.')
+    def test_get_without_identifier_with_invalid_page(self, rf):
+        factories.QueueEntryFactory.create_batch(30)
+        request = rf.get('/', {'page': 400}, HTTP_HOST='example.com')
+        response = views.queue(request)
+        assert response.status_code == 200
+
+    def test_delete(self, rf):
+        entry = factories.QueueEntryFactory.create()
+        request = rf.delete('/')
+        response = views.queue(request, entry.ark)
+
+        assert response.status_code == 200
+        assert QueueEntry.objects.count() == 0
+
+    def test_delete_returns_not_found(self, rf):
+        request = rf.delete('/')
+        response = views.queue(request, 'ark:/000001/dne')
+        assert response.status_code == 404
+
+    def test_put(self, queue_xml, rf):
+        entry = factories.QueueEntryFactory.create(ark='ark:/67531/coda4fnk', bytes='4')
+
+        request = rf.put('/', queue_xml, 'application/xml', HTTP_HOST='example.com')
+        response = views.queue(request, entry.ark)
+
+        updated_entry = QueueEntry.objects.get(ark=entry.ark)
+
+        assert entry.oxum != updated_entry.oxum
+        assert response.status_code == 200
+        assert response['Content-Type'] == 'application/atom+xml'
+
+    @pytest.mark.xfail(reason="The PUT method with out an identifier should not be allowed.")
+    def test_put_without_identifier(self, queue_xml, rf):
+        factories.QueueEntryFactory.create(ark='ark:/67531/coda4fnk', bytes='4')
+        request = rf.put('/', queue_xml, 'application/xml', HTTP_HOST='example.com')
+        response = views.queue(request)
+
+        assert response.status_code == 404
```

Comment:
hard to say. Some of this stuff will likely change quite a bit. Ill add it in for now at least